profile-widget/qmlprofile.cpp

```diff
@@ -24,9 +24,18 @@ QMLProfile::QMLProfile(QQuickItem *parent) :
 	m_profileWidget->setPrintMode(true);
 	m_profileWidget->setFontPrintScale(fontScale);
 	connect(QMLManager::instance(), &QMLManager::sendScreenChanged, this, &QMLProfile::screenChanged);
+	connect(this, &QMLProfile::scaleChanged, this, &QMLProfile::triggerUpdate);
 	setDevicePixelRatio(QMLManager::instance()->lastDevicePixelRatio());
 }
 
+// we need this so we can call update() (which eventually redraws the profile and calls paint()
+// from the scaleChanged() signal -- this way as we scale out we can ensure that we shift the offsets
+// to not show "empty space" outside the profile when panning
+void QMLProfile::triggerUpdate()
+{
+	emit update();
```

Comment:
According to the docs, update is not a signal and therefore the `emit` is conceptually wrong here:
https://doc.qt.io/qt-5/qquickpainteditem.html

Practically, I believe `emit` is simply replaced by the empty string. However, some static checkers may complain.

Reply:
oops - I can't even tell why I was confused about this... you are right.